frontend/src/common.ts

```diff
@@ -752,19 +752,55 @@ export interface Class {
     events: ProgramClassEvent[];
     created_at: Date;
 }
-
-export interface ResidentProgramClassInfo {
+export interface ResidentProgramOverview {
     program_name: string;
     class_name: string;
     status: ProgClassStatus;
-    enrollment_status: EnrollmentStatus;
+    credit_types: string;
+    program_id: number;
+    class_id: number;
+    updated_at: string;
+
+    // enrollment info-only
+    enrollment_status?: EnrollmentStatus;
     start_date: string;
-    end_date: string;
-    attendance_percentage: string;
+    end_date?: string;
+    present_attendance?: number;
+    absent_attendance?: number;
+    attendance_percentage?: number;
+}
+
+export interface CalendarEvent {
+    event_id: number;
     class_id: number;
-    program_id: number;
+    program_name: string;
+    start_time: string;
+    duration: number;
+    location: string;
+    is_cancelled: boolean;
+}
+export interface StudentCalendar {
+    day_index: number;
+    date: string;
+    events: CalendarEvent[];
 }
 
+export interface StudentCalendarResponse {
+    data: {
+        days: StudentCalendar[];
+    };
+}
+export interface DayData {
+    date: string;
+    events: CalendarEvent[];
+}
+export function mapStudentCalendarResponseToDayData(
+    resp: StudentCalendarResponse | undefined
+): DayData[] {
+    // if resp is undefined or resp.data.days isn’t there, return empty array
+    const days = resp?.data?.days ?? [];
+    return days.map(({ date, events }) => ({ date, events }));
+}
```

Comment:
@corypride Is this something we can move to the page or does it need to exist here?